src/geocat/comp/climatology.py

```diff
@@ -322,3 +352,136 @@ def month_to_season(
     # Therefore, we must filter out the NaNs
     return means.sel(
         {time_coord_name: means[time_coord_name].dt.month == months[1]})
+
+
+def calendar_average(
+        dset: typing.Union[xr.DataArray, xr.Dataset],
+        freq: str,
+        time_dim: str = None,
+        climatology: bool = False,
+        calendar: str = 'standard') -> typing.Union[xr.DataArray, xr.Dataset]:
```

Comment:
Could we set this `None` by default,
```suggestion
        calendar: str = None) -> typing.Union[xr.DataArray, xr.Dataset]:
```

and then retrieve it from the `attrs` or `encoding` of the input dataset?

```python
In [15]: ds.time.encoding
Out[15]: 
{'source': '/Users/abanihi/Library/Caches/xarray_tutorial_data/eee06791cc19e59f074155a82a7ffe90-rasm.nc',
 'original_shape': (36,),
 'dtype': dtype('float64'),
 'units': 'days since 0001-01-01',
 'calendar': 'noleap'}

In [16]: ds.time.attrs
Out[16]: {'long_name': 'time', 'type_preferred': 'int'}
```

Reply:
no, you should figure it out from `ds.time[0]` (see `infer_calendar_name` here: https://github.com/pydata/xarray/pull/5233/files#diff-e1c5f34f91e5b8a5242d9499ca9ff2765f3d5b7956c28196f80e4c399b29b1a6L325-R362) `attrs` and `encoding` record how the information was stored on disk. It could be totally out-of-date (unlikely but possible).

Reply:
@andersy005 I agree that the calendar should be retrieved from the data, but if I do this, would there be a need to keep the `calendar` kwarg? It seems redundant.

@dcherian `infer_calendar_name` returns the correct strings, so I can definitely use this approach.

Reply:
@andersy005 I have added your suggestion and updated the doc string. Please let me know if it needs to be changed.